R/textstat_dist.R

```diff
@@ -134,39 +97,37 @@ textstat_dist <- function(x, selection = character(0), n = NULL,
         result <- x
     }
     
-    # truncate to n if n is not NULL
     if (!is.null(n))
-        result <- head(as.matrix(result), n)
+        result <- result[1:n,]
     
-    # discard the upper diagonal if tri == TRUE
-    if (tri)
-        result[upper.tri(result, diag = !diag)]<-0
+    # # discard the upper diagonal if upper == FALSE
+    # if (!upper)
+    #     result <- Matrix::tril(result)
     
-    if (dist){
-        # create a new dist object
-        p <- nrow(result)
-        if(ncol(result) != p) warning("non-square matrix")
-        
-        
-        # only retain lower triangular elements for the dist object
-        distM <- result[row(result) > col(result)]
-        
-        # set the attributes of the dist object
-        attributes(distM) <- NULL
-        attr(distM, "Size") <- nrow(result)
-        if (!is.null(rownames(result))) attr(distM, "Labels") <- rownames(result)
-        attr(distM, "Diag") <- diag
-        attr(distM, "Upper") <- !tri
-        attr(distM, "method") <- method
-        attr(distM, "call") <- match.call()
-        attr(distM, "dimnames") <- NULL
-        class(distM) <- "dist"
-        
-        # This will call Stats::print.dist() and Stats::as.matrix.dist()
-        distM
-    } else {
-        result
-    }
+    
+    # # create a new dist object
+    # p <- nrow(result)
+    # if(ncol(result) != p) warning("non-square matrix")
+    # 
+    # 
+    # # only retain lower triangular elements for the dist object
+    # distM <- result[row(result) > col(result)]
+    # 
+    # # set the attributes of the dist object
+    # attributes(distM) <- NULL
+    # attr(distM, "Size") <- nrow(result)
+    # if (!is.null(rownames(result))) attr(distM, "Labels") <- rownames(result)
+    # attr(distM, "Diag") <- diag
+    # attr(distM, "Upper") <- upper
+    # attr(distM, "method") <- method
```

Comment:
Note: You can still add the additional attributes (such as "method") after the as.dist() call